src/controlnet_aux/canny/__init__.py

```diff
@@ -9,12 +9,11 @@ def __call__(self, img, low_threshold, high_threshold):
         input_type = "np"
         if isinstance(img, Image.Image):
             img = np.array(img)
-            input_type = "pil"
         
         img = HWC3(img)
         img = cv2.Canny(img, low_threshold, high_threshold)
         
-        if input_type == "pil":
-            img = Image.fromarray(img)
+        img = Image.fromarray(img)
+        img = img.convert("RGB")
             
         return img
```

Comment:
Now we cannot return a numpy array anymore no?

Reply:
yes. the intention for this PR is to reduce data type error and channel error when using diffuser pipeline. do you like it to return numpy array?

Reply:
Just in terms of backwards breaking compatibility I think it would be good to make sure things that previously worked still work, could we maybe still allow numpy?

Reply:
okay. i change it to return 'rgb' mode when it is pil then there will be no channel error with diffusers. revert to support numpy